interface/wx/httprequest.h

```diff
@@ -0,0 +1,232 @@
+// Name:        httprequest.h
+// Created:     2018-10-14
+// Copyright:   (c) 2018 wxWidgets development team
+// Licence:     wxWindows licence
+/////////////////////////////////////////////////////////////////////////////
+
+/**
+    @class wxHTTPRequest
+
+    wxHTTPRequest allows for simple HTTP requests using the operating systems
+    components as implementation. The latest features of their operating
+    system will be used if available.
+
+    This class is well suited for simple HTTP requests.
+
+    For example:
+
+    @code
+        wxHTTPRequest request("http://www.wxwidgets.org/downloads/logos/blocks.png");
+        if (request.Send())
+        {
+            // Write the response data to file
+            wxInputStream* inputStream = request->GetResponseStream();
+            wxFileOutputStream outputStream("blocks.png");
+            outputStream.Write(*inputStream);
+        }
+        else
+        {
+            // Log the error
+            wxLogError("Could not load logo: %s", request.GetStatusText());
+        }
+    @endcode
+
+    System wide configuration like proxy and SSL certificates will be used
+    when possible.
+
+    @section descriptions Implementation Descriptions
+
+    The following APIs are used per platform, additional documentation
+    about supported features may be found in ther documentation
+
+    @par WinHTTP on Windows
+
+    This API is available on all supported Windows Versions and supports
+    HTTP and HTTPs up to HTTP/2.
+
+    <a href="https://docs.microsoft.com/en-us/windows/desktop/WinHttp/about-winhttp">
+    API Documentation</a>
+
+    @par NSURLSession on macOS and iOS
+
+    This API is available since macOS 10.9 and iOS 7.0.
+
+    <a href="https://developer.apple.com/documentation/foundation/urlsession">
+    API Documentation</a>
+
+    @par libcurl on Linux
+
+    This library is available on most Unix distributions.
+
+    <a href="https://curl.haxx.se/libcurl/">
+    API Documentation</a>
+
+    @since 3.1.2
+    @library{wxnet}
+    @category{net}
+
+    @see wxHTTPSession
+*/
+class wxHTTPRequest
+{
+public:
+    /**
+        Create a new request instance
+
+        @param url The URL of the HTTP resource for this request
+        @param session If no wxHTTPSession is give the default session will
+                       be used
+
+        @since 3.1.2
+    */
+    wxHTTPRequest(const wxString& url, wxHTTPSession session = NULL);
+
+    /**
+
+        @since 3.1.2
+    */
+    ~wxHTTPRequest();
+
+    /**
+        Adds a request header send by this request.
+
+        @param name Name if the header
+        @param value String value of the header
+
+        @since 3.1.2
+    */
+    void AddRequestHeader(const wxString& name, const wxString& value);
+
+    /**
+        Set HTTP method.
+
+        Set <a href="http://www.w3.org/Protocols/rfc2616/rfc2616-sec9.html">common</a>
+        or expanded HTTP method.
+
+        Overrides GET or POST methods that is used by default.
```

Comment:
How is the default method selected?

Reply:
Copied that from wxHTTP, but I think GET for everything that doesn't set SetPostData()